storage/flux/table.gen.go.tmpl

```diff
@@ -742,40 +743,38 @@ func (t *{{.name}}GroupTable) advance() bool {
 		return true
 	}
 
-	// handle the group with aggregate case
+	var timestamps []int64
+	var values []{{.Type}}
+	var timestamp int64
 	var value {{.Type}}
-	// For group count, sum, min, and max, the timestamp here is always math.MaxInt64.
-	// their final result does not contain _time, so this timestamp value can be anything
-	// and it won't matter.
-	// For group first, we need to assign the initial value to math.MaxInt64 so
-	// we can find the row with the smallest timestamp.
-	// Do not worry about data with math.MaxInt64 as its real timestamp.
-	// In OSS we require a |> range() call in the query and a math.MaxInt64 timestamp
-	// cannot make it through.
-	var timestamp int64 = math.MaxInt64
-	if t.gc.Aggregate().Type == datatypes.AggregateTypeLast {
-		timestamp = math.MinInt64
-	}
+	{{if and (ne .Name "Boolean") (ne .Name "String")}}
+	var next {{.Type}}
+	{{end}}
+
 	for {
-		// note that for the group aggregate case, len here should always be 1
-		for i := 0; i < len; i++ {
-			switch t.gc.Aggregate().Type {
-			case datatypes.AggregateTypeCount:
-				{{if eq .Name "Integer"}}fallthrough{{else}}panic("unsupported for aggregate count: {{.Name}}"){{end}}
-			case datatypes.AggregateTypeSum:
-				{{if or (eq .Name "String") (eq .Name "Boolean")}}panic("unsupported for aggregate sum: {{.Name}}"){{else}}value += arr.Values[i]{{end}}
-			case datatypes.AggregateTypeFirst:
-				if arr.Timestamps[i] < timestamp {
-					timestamp = arr.Timestamps[i]
-					value = arr.Values[i]
-				}
-			case datatypes.AggregateTypeLast:
-				if arr.Timestamps[i] > timestamp {
-					timestamp = arr.Timestamps[i]
-					value = arr.Values[i]
-				}
-			}
+		groupBy, err := determine{{.Name}}AggregateMethod(t.gc.Aggregate().Type)
+		if err != nil {
+			panic(err)
 		}
+
+		length := arr.Size()
+		for i := 0; i < length; i++ {
+			t, v := groupBy(arr.Timestamps, arr.Values)
+			timestamps = append(timestamps, t)
+			values = append(values, v)
+		}
+
+		{{if and (ne .Name "Boolean") (ne .Name "String")}}
+		timestamp, next = groupBy(timestamps, values)
+		//todo(faith): trying to aggregate values between array cursors if the subsequent
+		// value is still part of the same group. Unsure of the logic here.
+		if len != 0 {
+			value += next
+		} else {
+			value = next
+		}
+		{{end}}
```

Comment:
I'm missing something in my logic here to preserve values between array cursors. I feel like the overall logic is pretty close, but there's something off here.